### PR TITLE
Feature 837 subframework for encrypting data at rest.

### LINKED ISSUE #837: Encrypt sechub configuration in database
## Situation
Currently the user configuration is not encrypted.

## Wanted 
We want user configuration data encrypted inside database. So user credentials inside configurations will be stored encrypted inside database

## Solution
- We introduce a new column inside Job table `JOB_CONFIG_ENCRYPTION` - per default it is `NONE` so we know which (old) job is NOT encrypted. 
- migrationscript uses `NONE` as default for database
- We introduce a new configuration value for sechub server (spring value) which contains a string with a secret key
- Before writing job data the job configuration will be encrypted by secret key, also the `JOB_CONFIG_ENCRYPTION` will be set
- Before returning job configuration object for admin rest calls we must decrypt, so admin can directly see the origin configuration by admin REST api

### LINKED ISSUE #3126: Provide a subframework to encrypt data at rest
### Info
Just for documentation purposes this issue was created with the provided details from the pull request #3105 

### Situation
We need the possibility to encrypt data at rest.

## Wanted (copied and adopted from related pull request)
Implement a sub-framework for encrypting data at-rest.
Designed to encrypt text data which needs to be stored into a persistent storage (data base, object storage, file etc.).

The algorithms used fulfill the following requirements:
- protect both confidentiality and integrity
- are nonce misuse-resistant

Allows to rotate:
- initialization vectors (nonces)
- secrets
- ciphers

### PR DESCRIPTION
**Pull request by Jeeppler**
Implements a sub-framework for encrypting data at-rest.

It is designed to encrypt text data which needs to be stored into a persistent storage (data base, object storage, file etc.).

The algorithms used fulfill the following requirements:

- protect both confidentiality and integrity
- are nonce misuse-resistant

Allows to rotate:

- initialization vectors (nonces)
- secrets
- ciphers

HINT: The actual database encryption as described in #837 and #838 is not implemented.

- closes #3126

### CODE REVIEW

File: sechub-commons-core/src/test/java/com/mercedesbenz/sechub/commons/core/security/persistence/AesGcmSivTest.java [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+        BinaryString secret = new Base64String("a".repeat(31));
+
+        /* execute + test */
+        assertThrows(InvalidKeyException.class, () -> {
+            AesGcmSiv.create(secret);
+        });
+    }
+
+    @Test
+    void encrypt__aes_256() throws InvalidKeyException, NoSuchAlgorithmException, NoSuchPaddingException, InvalidAlgorithmParameterException,
```
Comment (lorriborri) [inline sechub-commons-core/src/test/java/com/mercedesbenz/sechub/commons/core/security/persistence/AesGcmSivTest.java:81]: Optional: Remove unnecessary Exceptions from throw list (in whole test class)

File: sechub-commons-core/src/test/java/com/mercedesbenz/sechub/commons/core/security/persistence/AesGcmSivTest.java [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+        /* test */
+        assertEquals(expectedCiphertext, ciphertext.toString());
+        assertEquals(PersistenceCipherType.AES_GCM_SIV_256, cipher.getCipherType());
+    }
+
+    @Test
+    void encrypt__aes_128_base64_format_and_emojis() throws InvalidKeyException, InvalidAlgorithmParameterException {
+        /* prepare */
+        BinaryString secret = new Base64String("🍐🍌🍓🍉");
+        ;
```
Comment (lorriborri) [inline sechub-commons-core/src/test/java/com/mercedesbenz/sechub/commons/core/security/persistence/AesGcmSivTest.java:197]:
```suggestion
      
```

**Review by lorriborri — APPROVED** [review]
Looks good to me :star2: Good work :+1: 
I've found some optional Issues

File: sechub-commons-core/src/main/java/com/mercedesbenz/sechub/commons/core/security/persistence/AesGcmSiv.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        return encrypt(plaintext, initializationVector, BinaryStringEncodingType.BASE64);
+    }
+
+    public String decrypt(BinaryString ciphertext, BinaryString initializationVector)
+            throws InvalidAlgorithmParameterException, InvalidKeyException, IllegalBlockSizeException, BadPaddingException {
+        Cipher cipher;
+        try {
+            cipher = Cipher.getInstance(ALGORITHM, cryptoProvider);
+
+            SecretKeySpec keySpec = new SecretKeySpec(secret.getEncoded(), "AES");
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-commons-core/src/main/java/com/mercedesbenz/sechub/commons/core/security/persistence/AesGcmSiv.java:98]: No, specific reason.

File: sechub-commons-core/src/main/java/com/mercedesbenz/sechub/commons/core/security/persistence/AesGcmSiv.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        return cipherType;
+    }
+
+    @Override
+    public BinaryString encrypt(String plaintext, BinaryString initializationVector, BinaryStringEncodingType encodingType)
+            throws InvalidAlgorithmParameterException, InvalidKeyException {
+        try {
+            Cipher cipher = Cipher.getInstance(ALGORITHM, cryptoProvider);
+
+            SecretKeySpec keySpec = new SecretKeySpec(secret.getEncoded(), "AES");
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-commons-core/src/main/java/com/mercedesbenz/sechub/commons/core/security/persistence/AesGcmSiv.java:127]: No, specific reason. You can optimize this afterwards.

File: sechub-commons-core/build.gradle [hunk 5]
```diff
@@ -3,6 +3,8 @@ dependencies {
     
     api spring_boot_dependency.slf4j_api
 
+    implementation 'org.bouncycastle:bcprov-jdk18on:1.73'
```
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-commons-core/build.gradle:6]: Please, do this after it is merged.

**Pull request merged**